### PR TITLE
3947/datetime input fixes

### PR DESCRIPTION
**Pull request by tatiana-scandi**
**Related issue(s):**
* Fixes calendar pop-up hidden below bottom bar in mobile view
* Fixes mobile keyboard activated on input click
* Fixes https://github.com/scandipwa/scandipwa/issues/3945
* Fixes https://github.com/scandipwa/scandipwa/issues/3947
* Fixes https://github.com/scandipwa/scandipwa/issues/3950

**In this PR:**
* Fixed mentioned bugs

### CODE REVIEW

**Comment by Elena-Evsa** [comment]
Issue 1: Passed
Issue 2: Passed
Issue 3: Passed
Issue 4: Passed 
Issue 5: Passed  
Issue 6: Please, remove 00 value from hours dropdown, it is mean the same as 12
image

Expected:
image

**Comment by tatiana-scandi** [comment]
Issue 6 fixed. Please check:
1. Minutes in select are in range 0 to 59.
2. Hours in select are in range 1 to 12 if time is AM/PM
3. Hours in select are in range 0 to 23 if time is 24h format
4. If format is AM/PM + time is 00:XX => hours autocomplete works + hours value is correct when product is added to cart. 
5. If format is AM/PM + time is 12:XX => hours autocomplete works + hours value is correct when product is added to cart. 
6. If format is AM/PM + you change hours to 00:XX => hours value is correct when product is added to cart. 
7.  If format is AM/PM + you change hours to 12:XX => hours value is correct when product is added to cart.
8. 4. If format is 24h + time is 00:XX => hours autocomplete works + hours value is correct when product is added to cart. 
9. If format is 24h + time is 12:XX => hours autocomplete works + hours value is correct when product is added to cart.
10.  If format is 24h + you change hours to 00:XX => hours value is correct when product is added to cart. 
11.  If format is 24 + you change hours to 12:XX => hours value is correct when product is added to cart. 

**No need to check option "Use JS calendar = YES".**

File: packages/scandipwa/src/util/Form/Extract.js [hunk 1]
```diff
@@ -120,6 +132,19 @@ export const getDateTimeFormat = (type, dateFieldsOrder, timeFormat) => {
     return `${datePart } ${ timePart}`.trim();
 };
 
+/** @namespace Util/Form/Extract/adjustAmpmHours */
+export const adjustAmpmHours = (hours, ampm) => {
+    if (ampm === 'PM') {
+        return (+hours % HOURS_12H_COUNT) + HOURS_12H_COUNT;
```
Comment (ejnshtein) [inline packages/scandipwa/src/util/Form/Extract.js:138]: If you want to convert string to number please use `Number.parseInt` or `parseInt` function. Using `+value` to convert is kind of a hack and can be confusing.

**Review by ejnshtein — CHANGES_REQUESTED** [review]
Please answer my comment!

File: packages/scandipwa/src/util/Form/Extract.js (hunk 1 above)
Comment (tatiana-scandi) [reply packages/scandipwa/src/util/Form/Extract.js:138]: It is a matter of what approach it is more welcome on each particular project. 

If we take ScandiPWA, string to number conversion via `+` is used 20+ times.
If you search for `parseInt`, it is used in ScandiPWA only once.

So I'm just using the approach most often used on the project.

Plus, from my personal preference is to use the shorter `+`.

Comment (ejnshtein) [reply packages/scandipwa/src/util/Form/Extract.js:138]: Still, using `+` is a hack and not everyone is familiar with it.
If it was used before it does not mean it should be used everywhere.

**Review by ejnshtein — APPROVED** [review]

**Pull request merged**